core/common/src/main/java/alluxio/util/URIUtils.java

```diff
@@ -50,6 +50,22 @@ public static URI appendPath(URI base, String path) throws URISyntaxException {
   }
 
   /**
+   * Appends the given path to the given base URI. It throws an {@link RuntimeException} if
+   * the inputs are malformed.
+   *
+   * @param base the base URI
+   * @param path the path to append
+   * @return the URI resulting from appending the base and the path
+   */
+  public static URI appendPathOrDie(URI base, String path) {
```

Comment:
I don't think we are taking advantage of URISyntaxException in our codebase.
This is because base and path are guaranteed to be valid.  I am fine with removing this if you prefer.